src/file.cpp

```diff
@@ -1950,34 +1957,31 @@ typedef struct _FILE_ALLOCATED_RANGE_BUFFER {
 			PTOKEN_PRIVILEGES PreviousState,
 			PDWORD ReturnLength);
 
-		static OpenProcessToken_t pOpenProcessToken = NULL;
-		static LookupPrivilegeValue_t pLookupPrivilegeValue = NULL;
-		static AdjustTokenPrivileges_t pAdjustTokenPrivileges = NULL;
-		static bool failed_advapi = false;
-		HMODULE advapi = NULL;
+		static BOOL result = -1;
+
+		if (result != -1) return !!result;
```

Comment:
I think it's better to use `bool` than `BOOL`, and I also think an explicit conversion to bool, `bool(X)` is better than the double negation trick `!!X`. But by just using the standard type, this conversion isn't necessary

Reply:
I'll replace `!!` with explicit cast to bool.